src/node/progress_tracker_types.h

```diff
@@ -93,18 +94,71 @@ namespace ccf
       new_views(ccf::Tables::NEW_VIEWS)
     {}
 
+    struct WriteBackupSigsMsg
+    {
+      WriteBackupSigsMsg(
+        ccf::BackupSignatures sig_value_, ProgressTrackerStoreAdapter* self_) :
+        sig_value(std::move(sig_value_)),
+        self(self_)
+      {}
+
+      ccf::BackupSignatures sig_value;
+      ProgressTrackerStoreAdapter* self;
+    };
+
     void write_backup_signatures(ccf::BackupSignatures& sig_value) override
+    {
+      if (threading::ThreadMessaging::thread_count > 1)
+      {
+        // When running on multiple threads a dead lock can occur between 2
+        // threads:
+        // Thread (a): a primary signature is emitted, first taking the
+        // version lock and then the progress tracker lock
+        // Thread (b): a backup signature is received which may result in the
+        // backup signatures being written to the ledger, first taking the
+        // progress tracker lock and then the version lock.
+        //
+        // We resolve this problem by writing the backup signatures
+        // asynchronously.
```

Comment:
There's probably something obvious going on here, but I'm failing to grasp it :)